classes/class-wc-rest-connect-shipping-rates-controller.php

```diff
@@ -18,13 +18,34 @@ class WC_REST_Connect_Shipping_Rates_Controller extends WC_REST_Connect_Base_Con
 	 */
 	public function post( $request ) {
 		$payload = $request->get_json_params();
+		$payload[ 'payment_method_id' ] = $this->settings_store->get_selected_payment_method_id();
 		$order_id = $request[ 'order_id' ];
 
 		// This is the earliest point in the printing label flow where we are sure that
 		// the merchant wants to ship from this exact address (normalized or otherwise)
 		$this->settings_store->update_origin_address( $payload[ 'origin' ] );
 		$this->settings_store->update_destination_address( $order_id, $payload[ 'destination' ] );
 
+		// Update the customs information on all this order's products
+		$updated_product_ids = array();
+		foreach ( $payload[ 'packages' ] as $package_id => $package ) {
+			if ( ! isset( $package[ 'contents_type' ] ) ) {
+				break; // No customs information in this order, bail
+			}
+			foreach ( $package[ 'items' ] as $index => $item ) {
+				if ( ! isset( $updated_product_ids[ $item[ 'product_id' ] ] ) ) {
+					$updated_product_ids[ $item[ 'product_id' ] ] = true;
+					update_post_meta( $item[ 'product_id' ], 'wc_connect_customs_info', array(
```

Comment:
Should we use or hook into the update_options work in https://github.com/Automattic/woocommerce-services/blob/6158b29c0dc3af4f5964942e507d2febf02ee143/classes/class-wc-connect-options.php so that these can be cleaned up neatly during uninstall?

Reply:
Ok! Sounds good to me. Great point about not wanting to clean up the customs info anyway!